### PR TITLE
bpf: add hybrid NodePort mode (DSR for TCP, SNAT for UDP)

### LINKED ISSUE #10079: NodePort: DSR with tcp, SNAT with UDP
Allow configuration of NodePort DSR only with TCP and have the rest via SNAT for UDP to avoid having to mess with MTU.

### PR DESCRIPTION
**Pull request by borkmann**
See commit msg. /cc @brb 

```release-note
Add a new DSR/SNAT hybrid mode which allows to work without MTU changes and enables DSR for TCP and SNAT for UDP workloads. Enable it by default for Cilium's kube-proxy replacement in probe and strict mode.
```

Fixes: #10079

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10203)

### CODE REVIEW

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

File: daemon/daemon_main.go [hunk 1]
```diff
@@ -1449,6 +1449,10 @@ func initKubeProxyReplacementOptions() {
 		log.Infof("Auto-enabling %q, %q, %q features", option.EnableNodePort,
 			option.EnableExternalIPs, option.EnableHostReachableServices)
 
+		if option.Config.Tunnel == option.TunnelDisabled {
+			option.Config.NodePortMode = option.NodePortModeHybrid
+		}
```
Comment (aanm) [inline daemon/daemon_main.go]: Maybe we should log this somehow? :thinking:

Comment (borkmann) [reply daemon/daemon_main.go]: Yeah, makes sense.

**Comment by borkmann** [comment]
test-me-please

**Review by brb — APPROVED** [review]
LGTM!

For a follow-up:

- As discussed offline, we should detect TCP Fast Open, and then switch to `snat` if detected.
- We should extend the k8sT/Services.go to validate the hybrid mode.

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
@nebril confirmed that `pr-nightly` in here should be ignored.

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

**Pull request merged**